system/lib/wasmfs/paths.cpp

```diff
@@ -170,4 +170,34 @@ ParsedFile getFileFrom(std::shared_ptr<Directory> base, std::string_view path) {
   return doParseFile(path, base, FollowLinks, recursions);
 }
 
+std::vector<std::string_view> splitPath(std::string_view path) {
```

Comment:
Perhaps we can move this (seemingly very generic) function into `emscripten_wget.c` along with the implementation of `emscripten_wget` itself?

Reply:
(this is now moved into another system library together with `emscripten_wget`)

Reply:
I'm concerned about the duplication of path parsing logic between this and `doParseParent`. Can we get away without this separate splitting logic, perhaps by looping over all the path separators in the input string in the implementation of `emscripten_wget` directly? In other words, can we implement this the same way we would if we didn't have access to WasmFS internals?

Reply:
What we really want here is recusrive mkdir helper.  How about just copying this one.. it looks very small: 
 https://stackoverflow.com/a/2336245.. that way we can avoid using C++ at all.

Reply:
@tlively This doesn't use any WasmFS internals since it was moved out in the last commits.

@sbc100 The current code is more clear than that link, I think? Maybe I just prefer C++...

Reply:
I think the C version is much simpler.   It takes way fewer lines, no?  And it won't bring in libc++ code so should be a *lot* smaller.

Also, it seems reasonable to be able to use emscripten_wget in C programs without linking libc++ (know once we switch -sWASMFS by default that won't be possible any more.

Reply:
Good point... for code size reasons we should do the simpler thing. I'll look into it.

Reply:
Done.